Scripts/Player.gd

```diff
@@ -77,10 +69,8 @@ func process_input(delta: float):
 		
 	toggle_sprint(Input.is_action_pressed("sprint"))
 	
-	input_vector = input_vector.normalized()
-	
-	direction += -cam_transform.basis.z * input_vector.y
-	direction += cam_transform.basis.x * input_vector.x
+	direction = cam_transform.basis * Vector3(input_vector.x, 0, input_vector.y)
```

Comment:
```suggestion
	direction = cam_transform.basis * Vector3(input_vector.x, 0.0, input_vector.y)
```
Two things I've picked up:
1. Godot 4 is stricter with literal types, `0` is always an `int` and never a `float`.
2. The player can't move when looking perfectly up or down.

Reply:
1. Thanks for pointing out! I'm relatively new to GDScript, especially when it comes to the newer version (been using Godot with C# so far), so I can mess up things like that.
2. I tried to figure this out but the code is pretty messy right now so I would much rather rewrite the entire class with the state pattern in mind while fixing this issue. Not sure if we should move to C# entirely at this point, as I've been trying out newer betas and it becomes more and more stable to be actually used more extensively.